fabric-common/lib/BlockDecoder.js

```diff
@@ -1140,496 +1242,395 @@ function decodeFabricOUIdentifier(proto_organizational_unit_identitfiers) {
 
 	return endorsement;
 }
 
-function decodeProposalResponsePayload(proposal_response_payload_bytes) {
+function decodeProposalResponsePayload(proposalResponsePayloadBuf) {
 	const proposal_response_payload = {};
-	const proto_proposal_response_payload = fabprotos.protos.ProposalResponsePayload.decode(proposal_response_payload_bytes);
-	proposal_response_payload.proposal_hash = proto_proposal_response_payload.getProposalHash().toBuffer().toString('hex');
-	proposal_response_payload.extension = decodeChaincodeAction(proto_proposal_response_payload.getExtension());
+	const proposalResponsePayloadProto = fabproto6.protos.ProposalResponsePayload.decode(proposalResponsePayloadBuf);
+	proposal_response_payload.proposal_hash = proposalResponsePayloadProto.proposal_hash;
+	proposal_response_payload.extension = decodeChaincodeAction(proposalResponsePayloadProto.extension);
 
 	return proposal_response_payload;
 }
 
-function decodeChaincodeAction(action_bytes) {
+function decodeChaincodeAction(chaincodeActionBuf) {
 	logger.debug('decodeChaincodeAction - start');
 	const chaincode_action = {};
-	const proto_chaincode_action = fabprotos.protos.ChaincodeAction.decode(action_bytes);
-	chaincode_action.results = decodeReadWriteSets(proto_chaincode_action.getResults());
-	chaincode_action.events = decodeChaincodeEvents(proto_chaincode_action.getEvents());
-	chaincode_action.response = decodeResponse(proto_chaincode_action.getResponse());
-	chaincode_action.chaincode_id = decodeChaincodeID(proto_chaincode_action.getChaincodeId());
+	const chaincodeActionProto = fabproto6.protos.ChaincodeAction.decode(chaincodeActionBuf);
+	chaincode_action.results = decodeReadWriteSets(chaincodeActionProto.results);
+	// it may be called events, however it is only one event at this time
+	chaincode_action.events = decodeChaincodeEvent(chaincodeActionProto.events);
+	chaincode_action.response = decodeResponse(chaincodeActionProto.response);
+	chaincode_action.chaincode_id = decodeChaincodeID(chaincodeActionProto.chaincode_id);
 
 	return chaincode_action;
 }
 
-function decodeChaincodeEvents(event_bytes) {
-	const events = {};
-	const proto_events = fabprotos.protos.ChaincodeEvent.decode(event_bytes);
-	events.chaincode_id = proto_events.getChaincodeId();
-	events.tx_id = proto_events.getTxId();
-	events.event_name = proto_events.getEventName();
-	events.payload = proto_events.getPayload().toBuffer();
+function decodeChaincodeEvent(chaincodeEventBuf) {
+	const event = {};
+	const chaincodeEventProto = fabproto6.protos.ChaincodeEvent.decode(chaincodeEventBuf);
+	event.chaincode_id = chaincodeEventProto.chaincode_id;
+	event.tx_id = chaincodeEventProto.tx_id;
+	event.event_name = chaincodeEventProto.event_name;
+	event.payload = chaincodeEventProto.payload;
 
-	return events;
+	return event;
 }
 
-function decodeChaincodeID(proto_chaincode_id) {
+function decodeResponse(responseProto) {
+	if (responseProto) {
+		const response = {};
```

Comment:
`
const {status,message,payload } = responseProto
return {status,message,payload }
`

Reply:
Yes we could do that, we should do that with an overall clean up of this class